src/_pytest/python_api.py

```diff
@@ -177,6 +313,51 @@ def __repr__(self) -> str:
             seq_type(self._approx_scalar(x) for x in self.expected)
         )
 
+    def repr_compare(
+        self, other_side: Sequence[float], verbosity_level: int
+    ) -> List[str]:
+        import math
+        import numpy as np
+
+        if len(self.expected) != len(other_side):
+            return [
+                "Impossible to compare lists with different sizes.",
```

Comment:
It would be good to utilize the length-comparison part of `_compare_eq_sequence` here. I guess it would need to be split from there.

(You don't have to do this, maybe add a TODO)